packages/headless/src/features/pagination/pagination-slice.ts

```diff
@@ -70,10 +70,22 @@ export const paginationReducer = createReducer(
       .addCase(executeSearch.fulfilled, (state, action) => {
         const {response} = action.payload;
         state.totalCountFiltered = response.totalCountFiltered;
+      })
+      .addMatcher(isFacetAction, (state, action) => {
+        console.log(action.type);
+        state.firstResult = getPaginationInitialState().firstResult;
       });
   }
 );
 
+function isFacetAction(action: AnyAction) {
+  return (
+    !action.type.includes('analytics') &&
+    (action.type.includes('facet') || action.type.includes('Facet')) &&
+    (action.type.includes('toggle') || action.type.includes('deselect'))
+  );
+}
```

Comment:
This is the function used for the matcher. I wanted to capture all possible facet toggle and deselect actions without capturing the analytic actions. I'm not sure if this was the best way to do it. I'm open to any other potential suggestions.

Reply:
I think this approach is too risky & hard to maintain. The original approach that imports all the actions & add cases for them might have more code, but as soon as a change is made (an action's name is changed,removed, etc.) Typescript will give us a compilation error, and we know the exact actions which resets the pagination

Reply:
Ok, I will add a case for each action then